leetcode/arai60/memo.md

```diff
@@ -0,0 +1,75 @@
+# 112. Path Sum
+* 問題: https://leetcode.com/problems/path-sum/
+* 言語: Python
+
+## Step1
+* 深さ優先探索（DFS）でノードを探索しながら、そのノードまでの和を情報として保持していく
+* スタックにノードオブジェクトとそのノードまでの和の組を積んでいく
+
+### 解答（AC）
+```py
+class Solution:
+    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
+        if root is None:
+            return False
+        
+        node_and_val = [(root, root.val, root.val)]
+
+        while node_and_val:
+            node, val, sum = node_and_val.pop()
```

Comment:
組み込みの `sum` を上書きしているので、避けたほうがいいですね。`path_sum` とするか、やむを得なければ `sum_` でしょうか。

https://peps.python.org/pep-0008/#descriptive-naming-styles
> single_trailing_underscore_: used by convention to avoid conflicts with Python keyword, e.g. :
`tkinter.Toplevel(master, class_='ClassName')`